profit/sur/gaussian_process.py

```diff
@@ -482,6 +480,7 @@ def optimize(self, fixed_sigma_n=False, eval_gradient=False, return_hess_inv=Fal
         self.hyperparameters['sigma_f'] = np.atleast_1d(opt_hyperparameters[last_idx])
         if not self.fixed_sigma_n:
             self.hyperparameters['sigma_n'] = np.atleast_1d(opt_hyperparameters[-1])
+        self.print_hyperparameters("Optimized")
```

Comment:
In the future, such input/output should not be done inside the fitting code to decouple interface from core.

Reply:
... one way to do this would be a proper event architecture, where the console (or web) UI can pass an event handler that receives a message when hyperparameters are optimized, together with their values. The display logic is then handled by the respective UI frontend.

Reply:
This is a very good idea. We should discuss if a general event-driven architecture should be one of our main focuses for v0.5.